purchase_receipt_percentage/models/purchase_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    receipt_percentage = fields.Float(
+        compute="_compute_receipt_percentage",
+        store=True,
+        help="Receipt percentage between 0% and 100%",
+    )
+    receipt_percentage_display = fields.Float(
+        compute="_compute_receipt_percentage",
+        store=True,
+        help="Technical field to be displayed in view. Its value is between"
+        " 0 and 1, the percentage widget will format the value properly",
+    )
+
+    @api.depends("order_line.receipt_percentage")
+    def _compute_receipt_percentage(self):
+        with_lines = self.filtered("order_line")
+        (self - with_lines).update(
+            {"receipt_percentage": 100, "receipt_percentage_display": 1}
+        )
+        for order in with_lines:
+            lines_percentages = order.order_line.mapped("receipt_percentage")
+            perc = sum(lines_percentages) / len(lines_percentages)
+            order.update(
+                {
+                    "receipt_percentage": perc,
+                    # NB: we divide by 100 because we want this field to be in
+                    # [0, 1], the percentage widget in the view will take care
+                    # of displaying it correctly
+                    "receipt_percentage_display": perc / 100,
+                }
+            )
```

Comment:
You could do a read_group with `receipt_percentage:avg`.
With it you'd let psql do the computation + you'd compute multiple records at once, which would be a huge time saver if the field is shown on a tree view/kanban view